src/React.purs

```diff
@@ -259,6 +259,18 @@ type ReactSpec props state eff =
   , componentWillUnmount :: ComponentWillUnmount props state eff
   }
 
+cmapSpec
+  :: forall props' props state eff
+   . (props -> props')
+  -> ReactSpec props' state eff
+  -> ReactSpec props Unit eff
+cmapSpec f sp = (spec unit renderFn) { displayName = sp.displayName <> "Mapped" }
```

Comment:
It does hide the original spec, but still one can add new ones to the parent spec.